tests/integration/devfile/cmd_devfile_url_test.go

```diff
@@ -229,6 +229,23 @@ var _ = Describe("odo devfile url command tests", func() {
 			Expect(output).ShouldNot(ContainSubstring(url1))
 		})
 
+		It("should error out when a host is provided with a route on a openShift cluster", func() {
+
+			if os.Getenv("KUBERNETES") == "true" {
```

Comment:
Shouldn't we handle this in JustBeforeEach ? Because using it inside the spec will completely destroy the meaning of test.

Reply:
I don't understand what you mean by handling it in JustBeforeEach(). Please elaborate.

Reply:
This is the same way other openshift specific tests have been written 
https://github.com/openshift/odo/blob/80affddedd7547bf8b4587765b939c53ca9f9c00/tests/integration/devfile/cmd_devfile_url_test.go#L303
https://github.com/openshift/odo/blob/80affddedd7547bf8b4587765b939c53ca9f9c00/tests/integration/devfile/cmd_devfile_url_test.go#L201
https://github.com/openshift/odo/blob/80affddedd7547bf8b4587765b939c53ca9f9c00/tests/integration/devfile/cmd_devfile_url_test.go#L130
If you have any better idea to arrange these tests, please open a new issue.

Reply:
This is very unfortunate that we missed those reference that you point here otherwise we could have categorized those test just by putting them in a separate context and of course the precondition of KUBERNETES cluster check in JustBeforeEach() teardown steps. 

Anyway I will categorize those test in a separate pr for a better spec structure.

Reply:
Thanks for the advice. I have moved all the openshift specific URL tests into a new context and implemented the skipping logic in a JustBeforeEach() for the context.